### PR TITLE
feat: add role flow

### PR DESCRIPTION
**Pull request by diogo-costa-bjss**
- Created add role config that works for multiple instances.
    - Add role from user information 
    - Add role from create user flow
- Created a reusable component for user information (when a user is found by email during create or add role flows)
- Improved alerts and redirect from create user and add role

### CODE REVIEW

File: src/modules/feature-modules/admin/pages/users/roles/role-new.config.ts [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+  | { role: AvailableRoles }
+  | {
+    role: UserRoleEnum.ACCESSOR | UserRoleEnum.QUALIFYING_ACCESSOR;
+    organisationId: string;
+    unitIds: string[];
+  };
+export type OutboundPayloadType = CreateRolesType;
+
+// consts.
+const injector = AppInjector.getInjector();
```
Comment (ricardo-jorge-tavares) [inline src/modules/feature-modules/admin/pages/users/roles/role-new.config.ts:58]: You don't need this I believe...

File: src/modules/feature-modules/admin/components/user-information.component.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+  USER_DETAILS = 'Go to user details',
+  ADD_USER = 'Add user',
+  CONTINUE = 'Continue'
+}
+
+@Component({
+  selector: 'app-admin-pages-user-information',
+  templateUrl: './user-information.component.html',
+})
+export class UserInformationComponent extends CoreComponent implements OnInit {
```
Comment (ricardo-jorge-tavares) [inline src/modules/feature-modules/admin/components/user-information.component.ts:21]: A component should NOT extend from CoreComponent.

Only "page" components should do that.

File: src/modules/feature-modules/admin/pages/users/roles/role-new.config.ts (hunk 1 above)
Comment (diogo-costa-bjss) [reply src/modules/feature-modules/admin/pages/users/roles/role-new.config.ts:58]: I need it to inject `authenticationStore`

Comment (ricardo-jorge-tavares) [reply src/modules/feature-modules/admin/pages/users/roles/role-new.config.ts:58]: Gotcha... ;)

**Review by ricardo-jorge-tavares — APPROVED** [review]

**Pull request merged**